### PR TITLE
Renaming of events fetcher and dispatcher

### LINKED ISSUE #522: renaming of the current classes / pulling fetcher to a separate package;

### PR DESCRIPTION
**Pull request by joke1196**
closes #522

### CODE REVIEW

**Review by travissarles — DISMISSED** [review]

**Review by travissarles — DISMISSED** [review]

File: event-log/src/main/scala/io/renku/eventlog/Microservice.scala [hunk 1]
```diff
@@ -93,12 +93,12 @@ object Microservice extends IOMicroservice {
                                                        ApplicationLogger
                                 )
         subscriptions <- Subscriptions(ApplicationLogger)
-        eventsDispatcher <- EventsDispatcher(transactor,
-                                             subscriptions,
-                                             waitingEventsGauge,
-                                             underProcessingGauge,
-                                             queriesExecTimes,
-                                             ApplicationLogger
+        eventsDispatcher <- EventsDistributor(transactor,
```
Comment (jachro) [inline event-log/src/main/scala/io/renku/eventlog/Microservice.scala:96]: Shouldn't the `eventsDispatcher` become `eventsDistributor`?

File: event-log/src/main/scala/io/renku/eventlog/subscriptions/newEvent/ProjectPrioritisation.scala [hunk 2]
```diff
@@ -16,7 +16,7 @@
  * limitations under the License.
  */
 
-package io.renku.eventlog.subscriptions
+package io.renku.eventlog.subscriptions.newEvent
```
Comment (jachro) [inline event-log/src/main/scala/io/renku/eventlog/subscriptions/newEvent/ProjectPrioritisation.scala:19]: What do you think if the package name would be smth like `newandrecoverable` or better?

File: event-log/src/main/scala/io/renku/eventlog/subscriptions/newEvent/ProjectPrioritisation.scala [hunk 3]
```diff
@@ -26,10 +26,10 @@ import ch.datascience.tinytypes.{BigDecimalTinyType, TinyTypeFactory}
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.numeric.NonNegative
 import io.renku.eventlog.EventDate
-import io.renku.eventlog.subscriptions.ProjectPrioritisation.{Priority, ProjectIdAndPath, ProjectInfo}
+import io.renku.eventlog.subscriptions.newEvent.ProjectPrioritisation.{Priority, ProjectIdAndPath, ProjectInfo}
 
-private class ProjectPrioritisation {
-  import Priority._
+private[subscriptions] class ProjectPrioritisation {
```
Comment (jachro) [inline event-log/src/main/scala/io/renku/eventlog/subscriptions/newEvent/ProjectPrioritisation.scala:31]: Why do we need to make it private for the `subscription`?

**Review by travissarles — APPROVED** [review]

File: event-log/src/test/scala/io/renku/eventlog/subscriptions/EventsDistributorSpec.scala [hunk 4]
```diff
@@ -378,7 +378,7 @@ class EventsDispatcherSpec extends AnyWordSpec with MockFactory with Eventually
     val statusUpdatesRunner  = mock[StatusUpdatesRunner[IO]]
     val eventsSender         = mock[EventsSender[IO]]
     val logger               = TestLogger[IO]()
-    val dispatcher = new EventsDispatcher(
+    val dispatcher = new EventsDistributor(
```
Comment (jachro) [inline event-log/src/test/scala/io/renku/eventlog/subscriptions/EventsDistributorSpec.scala]: `distributor` maybe?

File: event-log/src/main/scala/io/renku/eventlog/subscriptions/newEvent/ProjectPrioritisation.scala (hunk 2 above)
Comment (joke1196) [reply event-log/src/main/scala/io/renku/eventlog/subscriptions/newEvent/ProjectPrioritisation.scala:19]: Ok good to know :D

File: event-log/src/main/scala/io/renku/eventlog/subscriptions/newEvent/ProjectPrioritisation.scala (hunk 3 above)
Comment (joke1196) [reply event-log/src/main/scala/io/renku/eventlog/subscriptions/newEvent/ProjectPrioritisation.scala:31]: Yes good catch it shouldn't be used outside of the newandrecoverable package

File: event-log/src/main/scala/io/renku/eventlog/Microservice.scala [hunk 5]
```diff
@@ -36,7 +36,7 @@ import io.renku.eventlog.latestevents.IOLatestEventsEndpoint
 import io.renku.eventlog.metrics._
 import io.renku.eventlog.processingstatus.IOProcessingStatusEndpoint
 import io.renku.eventlog.statuschange.IOStatusChangeEndpoint
-import io.renku.eventlog.subscriptions.{EventsDispatcher, IOSubscriptionsEndpoint, Subscriptions}
+import io.renku.eventlog.subscriptions.{EventsDistributor, IOSubscriptionsEndpoint, Subscriptions}
```
Comment (jachro) [inline event-log/src/main/scala/io/renku/eventlog/Microservice.scala:39]: Do we also want to rename `Subscriptions` class to `Subscribers` (or smth better)?

Comment (joke1196) [reply event-log/src/main/scala/io/renku/eventlog/Microservice.scala]: Yes correct!

File: event-log/src/main/scala/io/renku/eventlog/subscriptions/newEvent/ProjectPrioritisation.scala (hunk 2 above)
Comment (joke1196) [reply event-log/src/main/scala/io/renku/eventlog/subscriptions/newEvent/ProjectPrioritisation.scala]: We went for unprocessed. As NewAndRecoverabelEventsFetcherImpl was getting long

File: event-log/src/main/scala/io/renku/eventlog/subscriptions/unprocessed/ProjectPrioritisation.scala [hunk 6]
```diff
@@ -28,7 +28,7 @@ import eu.timepit.refined.numeric.NonNegative
 import io.renku.eventlog.EventDate
 import io.renku.eventlog.subscriptions.unprocessed.ProjectPrioritisation.{Priority, ProjectIdAndPath, ProjectInfo}
 
-private class ProjectPrioritisation {
+private[unprocessed] class ProjectPrioritisation {
```
Comment (jachro) [inline event-log/src/main/scala/io/renku/eventlog/subscriptions/unprocessed/ProjectPrioritisation.scala:31]: I believe if the Prioritisation class needs to private for the `unprocessed` package we could drop `[unprocessed]`

Comment (joke1196) [reply event-log/src/main/scala/io/renku/eventlog/subscriptions/unprocessed/ProjectPrioritisation.scala:31]: We can make it work if we move around the ProjectIdAndPath case class. Maybe in a package

**Review by jachro — APPROVED** [review]
👍

**Pull request merged**